keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -254,12 +263,138 @@ tried this feature in Alpha, we would have time to fix issues.
+to a very high number.
+
+###### What specific metrics should inform a rollback?
+The customer should look at the behavior of the maxUnavailable and if the rollout
+is not respecting the maxUnavailable, the feature should be rolledback.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+No
+
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+If their StatefulSet rollingUpdate section has the field maxUnavailable specified with
+a value different than 1.
+The below command should show maxUnavailable value:
+```
+kubectl get statefulsets -o yaml | grep maxUnavailable
+```
+
+###### How can someone using this feature know that it is working for their instance?
+Ensure the number of replicas in your StatefulSet is more than 2. 
+Set maxUnavailable to a value 2 or more and then trigger a rolling update. 
+PMP Parallel:
+If you see the number of pods equal to maxUnavailable getting deleted at the same time,
+this means the feature is working as expected. After maxUnavailable pods are deleted,
+you will see a burst of pods equal to the number deleted up, start to come up
+
+PMP OrderedRedy:
+If you see the number of pods equal to maxUnavailable getting deleted at the same time,
+this means the feature is working as expected. After maxUnavailable pods are deleted,
+you will see a new pod come up(without violating ordering) and when it becomes ready,
+another will come up till we have all pods as ready that were deleted.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+controller-manager 
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+This feature when set, makes an extra loop over the total pods that belong
+to the StatefulSet.  It doesnt make any extra API calls.
+
+###### Will enabling / using this feature result in introducing new API types?
+This feature adds a new field in the StatefulSet RollingUpdate, called maxUnavailable.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+No
+
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+A new field gets added to every StatefulSet object which occupies 32bits.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+The increasing time per StatefulSet for rolling updates should be negligible.
```

Comment:
We don't have any SLOs that cover statefulsets.

So the answer is basically "no"

Reply:
yes